### DISCUSSION TITLE
[OP Stack Deployment] Economics of OP Stack

### DISCUSSION THREAD

**Discussion by ub-tech** [ethereum-optimism/developers#551] [Deploying the OP Stack] [answered]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Hi -- would appreciate direction on cost of running OP Stack chain (questions are below). 

**COST ESTIMATES**
**State Commitment** -- assume every 15mins and 0.004ETH/txn = 11.5 ETH/month (approx.)
**Data Availability** -- assume every 3minutes and 0.0001ETH/txn = 1.5 ETH/month (approx.)
**Total ETH/month = 13ETH**

**Compute Costs** (no other infra) = 4 high-CPU machines (32GB RAM) = $0.75/hr on GCP = $2k/month

**Total Costs = $47k/month**

### QUESTIONS:
(1) Are the estimates above correct?  Does anything change for joining the Superchain?

(2) Will this cost change based on gas/second being processed on the L2?  
Meaning, does the state output commitment require more gas with more L2-txns?  Similar question for the Data Availability costs (blobs are much cheaper so probably the ~420k gas per blob commit on L1 (which is the current OP Mainnet cost)).

(3) Any way to set sequencer to retain any L2 gas spent on transactions?

(4) Any L1 gas estimates for using "EVM-Ordered Alternative DA"? ([link](https://docs.optimism.io/builders/chain-operators/hacks/data-availability#evm-ordered-alternative-da)) Sounds like could savings since only the hash of the channel ([link](https://docs.optimism.io/stack/protocol/rollup/transaction-flow)) is posted to the L1.
 
(5) What configuration to manage how often the proposer submits to State Commitments to the L1?

Thank you very much.

**Comment by hash1go** [comment, marked as answer]
Hello!

1. L1 costs vary based on chain congestion. Refer to existing chain performance on [dashboards like Dune](https://dune.com/optimismfnd/optimism-l1-batch-submission-fees-security-costs).
Enabling fault proofs requires bonds for output proposals. See [this document](https://docs.optimism.io/stack/protocol/fault-proofs/explainer#how-much-eth-should-a-chain-operator-put-aside-to-operate-the-fault-proof-system) for more details.

2. When L2 is congested, more frequent data postings are required.

3. The sequencer retains profits from L2 transactions after deducting L1 costs. For more details, see [this specification](https://specs.optimism.io/protocol/exec-engine.html#fees).

4. I do not have data on the costs when using AltDA. You may refer to existing chain performance on Dune dashboards for insights.

5. The parameter "[proposal-interval](https://docs.optimism.io/builders/chain-operators/configuration/proposer#proposal-interval)" seems to be related, but honestly, I am not very familiar with changing the output proposal frequency.
